pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -3569,3 +3569,126 @@ func TestGetWorkloadNameForPod(t *testing.T) {
 		t.Errorf("Expected different workload name\n want: %s\n got: %s", wantWlName, wlName)
 	}
 }
+
+func TestReconciler_DeletePodAfterTransientErrorsOnUpdateOrDeleteOps(t *testing.T) {
```

Comment:
To avoid these conflicts, I will cherry-pick #1721 to release-0.6.

Reply:
Uhm, the `TestReconciler_DeletePodAfterTransientErrorsOnUpdateOrDeleteOps` could not be performed in the release-0.6 branch without this PR. (https://github.com/kubernetes-sigs/kueue/pull/2090)

So, could we just drop the `TestReconciler_DeletePodAfterTransientErrorsOnUpdateOrDeleteOps` from this cherry-pick PR?
@alculquicondor WDYT?

Reply:
Done

Reply:
@alculquicondor WDYT?

Reply:
I'm ok with removing the test.

/approve

@tenzen-y anything to add?

Reply:
Everything looks good to me.
Thanks!
/lgtm